ktor-client/ktor-client-features/ktor-client-json/common/src/io/ktor/client/features/json/JsonFeature.kt

```diff
@@ -65,11 +65,30 @@ class JsonFeature internal constructor(
          * Please note that wildcard content types are supported but no quality specification provided.
          */
         @KtorExperimentalAPI
-        var acceptContentTypes: List<ContentType> = listOf(ContentType.Application.Json)
+        var acceptContentTypes: MutableList<ContentType> = mutableListOf(ContentType.Application.Json)
+            @Deprecated(
+                "Use accept() function instead.",
+                replaceWith = ReplaceWith("accept(newList, onlyThese = true)")
+            )
             set(newList) {
                 require(newList.isNotEmpty()) { "At least one content type should be provided to acceptContentTypes" }
                 field = newList
             }
+
+        /**
+         * Adds accepted content types. Be aware that [ContentType.Application.Json] is accepted by default. Depending
+         * on the given [onlyThese] property previously added content types will be removed or will remain.
+         * It also affects `Accept` request header value.
+         */
+        @KtorExperimentalAPI
```

Comment:
Well, still there are issues with this. The first one is that one may accept the same content type multiple times. Shouldn't we use a set instead of the list or a check before adding? Also, the flag makes this function order-sensitive. Consider the following example:

```kotlin
install(JsonFeature) {
    accept(ContentType.Application.OctetStream, onlyThese = true) 
    accept(OctetStream.Application.Zip)
    accept(OctetStream.Application.FormUrlEncoded)
    accept(OctetStream.Application.ProblemXml)
}
```

So what if I want to simply reorder lines (to make it look "more beautiful", sort by length or something)? What is a change of that I forget about `onlyThese`? This `onlyThese` could be only at the beginning and makes no sense in the middle or at the last accept invocation, right?

Reply:
I added `onlyThese` to cover the case when we need to remove the default content type. Well, let's think about how we can avoid it.

Reply:
We may always apply the flag at the first accept invocation.

```kotlin
install(JsonFeature) { // only application/octet-stream
   accept(ContentType.Application.OctetStream)
}
```

```kotlin
install(JsonFeature) // the default application/json
```

```kotlin
install(JsonFeature) {
   accept(ContentType.Application.Json)
   accept(ContentType.Application.OctetStream)
}
```

Reply:
Good point, done.